### PR TITLE
Fix warning: Kernel#open is deprecated

### PR DESCRIPTION
**Pull request by mttkay**
## Description

Opening URIs by redirecting through `Kernel#open` is deprecated as of Ruby 2.7 and is dropped in Ruby 3.0.

To unblock applications that use Ruby 3, we need to replace these calls with `URI.open`.

See also:

* https://rubyreferences.github.io/rubychanges/2.7.html#network-and-web
* https://github.com/ruby/open-uri/commit/53862fa35887a34a8060aebf2241874240c2986a

## Related Issue

n/a

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [-] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

**Comment by mttkay** [comment]
Build failure look unrelated:

> STDERR: /opt/asdf/installs/ruby/2.7.2/lib/ruby/gems/2.7.0/gems/bundler-2.2.4/lib/bundler/spec_set.rb:87:in `block in materialize': Could not find ffi-1.10.0 in any of the sources (Bundler::GemNotFound)

It is also happening in other MRs, e.g.: https://buildkite.com/chef-oss/chef-license-scout-master-verify/builds/140#f51d32ff-3efe-4342-9c6b-dc773e962cc4

**Comment by mttkay** [comment]
@tduffield Could you or another maintainer do a code review and assist with fixing the build? Build failures do not look related to recent changes in pull requests.

**Comment by mttkay** [comment]
There are also 3 tests that fail locally after a fresh checkout of the project when on `master`:

```
$ bundle exec rspec
...
Finished in 5.1 seconds (files took 0.3487 seconds to load)
157 examples, 3 failures

Failed examples:

rspec ./spec/license_scout/dependency_manager/bundler_spec.rb:134 # LicenseScout::DependencyManager::Bundler#dependencies bundler 2.x project returns an array of Dependencies found in the directory
rspec ./spec/license_scout/dependency_manager/cargo_spec.rb:86 # LicenseScout::DependencyManager::Cargo#dependencies cargo project returns an array of Dependencies found in the directory
rspec ./spec/license_scout/dependency_manager/mix_spec.rb:80 # LicenseScout::DependencyManager::Mix#dependencies returns an array of Dependencies found in the directory
```

[testoutput.log](https://github.com/chef/license_scout/files/6775347/testoutput.log)

Will break out a separate issue for this.

**Comment by mttkay** [comment]
> Will break out a separate issue for this.

https://github.com/chef/license_scout/issues/265

**Comment by mttkay** [comment]
> @tduffield Could you or another maintainer do a code review and assist with fixing the build? Build failures do not look related to recent changes in pull requests.

Or @danielsdeleo maybe?

**Comment by tduffield** [comment]
@mttkay Apologies for the delay; I'm out on PTO this week. I'll take a look at this first thing when I get back (next Tuesday).

**Comment by mttkay** [comment]
> @mttkay Apologies for the delay; I'm out on PTO this week. I'll take a look at this first thing when I get back (next Tuesday).

No worries! I happen to be out on PTO myself all of next week -- enjoy your time off and thanks for looking into it!

**Comment by tduffield** [comment]
Okay, I've fixed the issue in the pipeline. You should be able to rebase on the latest `master` branch.

**Comment by mttkay** [comment]
Excellent, thanks! I am traveling all day today, but should be able to get to this some time tomorrow :+1:

**Comment by mttkay** [comment]
Thanks @tduffield, build is green now.

**Review by tduffield — APPROVED** [review]

**Comment by mttkay** [comment]
Thank you @tduffield for including this in 2.6.2 and tagging a new release. I am not familiar with the release cycle of Chef, but I noticed that the latest version on rubygems.org is still 2.5.1: https://rubygems.org/gems/license_scout

To adopt the new version, would we have to wait for a new Chef release, or is it possible to deploy 2.6.2 of `license_scout` to rubygems.org independently?

**Comment by nkierpiec** [comment]
@mttkay the new ruby gem (2.6.2) exists now

**Comment by mttkay** [comment]
Thanks @nkierpiec -- sorry to be a bother, but I only now realized that chef-omnibus actually pins `license_scout` to `1.x`, so I cannot declare a dependency on 2.6.2. We're on omnibus 7, but even on master this is still pinned to 1.x+: https://github.com/chef/omnibus/blob/master/omnibus.gemspec#L31

Should this be `~> 2.0`, since otherwise, no one will ever pull in the 2.x releases via a `bundle update`? :thinking: 

I'm not very familiar with the chef ecosystem of libraries and dependencies so I could be missing something.

**Comment by mttkay** [comment]
To illustrate:

```
Fetching https://dev.gitlab.org/gitlab/omnibus.git
Fetching gem metadata from https://rubygems.org/.........
Resolving dependencies...
Bundler could not find compatible versions for gem "license_scout":
  In Gemfile:
    license_scout (~> 2.6.2)

    omnibus was resolved to 7.0.10, which depends on
      license_scout (~> 1.0)
```

**Comment by nkierpiec** [comment]
Hey, @mttkay, not a bother, I am going to have to see if Tom has more insight on this since he has worked more closely with this product. @tduffield is this something you can clarify for @mttkay? Thanks!

**Comment by tduffield** [comment]
@mttkay There are changes required to Omnibus in order to have it support LicenseScout 2.x. If you're having issues with 1.x, you can open a PR against the 1-stable branch, which is our release branch for 1.x.

**Comment by mttkay** [comment]
Thanks all. I consulted with a co-worker as well, and it looks like we use a fork of Omnibus 7.0 so I was able to test these changes against our own repo; however, just requiring the newer licsense_scout version did not work because I ran into dependency errors with `mixlib-shellout` between what `chef`, `omnibus` and `license_scout` think it should be (`chef` and `omnibus` require or allow for 3.x, while `license_scout` pins it to 2.2).

I am now testing if we can relax that pin here: https://github.com/chef/license_scout/pull/269

Not sure I'm venturing in the right direction with this but I hope this will fix the problem.

**Pull request merged**
